Add OP Public Consumer

benjay10 opened the pull request:
**[DL-6212]**

As a response to https://github.com/lblod/app-digitaal-loket/pull/644, https://github.com/lblod/app-digitaal-loket/pull/648, and https://github.com/lblod/app-organization-portal/pull/523

This adds a consumer to ingest important data from OP needed for this application, instead of via Loket. This means switching to the new type of consumer with SPARQL mappings.

This also includes a migration to remove data that was previously ingested from Loket to get a clean start.

**For testing:** you should be able to start the application with attention to the migrations. After that, enable the consumer with the following settings:

```yaml
op-public-consumer:
  environment:
    DCR_DISABLE_INITIAL_SYNC: "false"
    DCR_DISABLE_DELTA_INGEST: "false"
    DCR_WAIT_FOR_INITIAL_SYNC: "true"
    DCR_SYNC_BASE_URL: "https://organisaties.abb.vlaanderen.be"
    DCR_CRON_PATTERN_DELTA_SYNC: '* * * * *' # this is every minute
    DCR_LANDING_ZONE_DATABASE: "database"
    DCR_REMAPPING_DATABASE: "database"
    SLEEP_BETWEEN_BATCHES: "1"
```

and watch as it ingests data. The `consumer` > `database` > `triplestore` throughput can be really slow while consuming 100% CPU. I can suggest to also look at [this PR to introduce the new `sparql-parser`](https://github.com/lblod/app-leidinggevenden-databank/pull/6). This works much faster. It changes "I'll let it run overnight and check in the morning" to "I'll quickly brew a cup of tea and check the results while sipping hot goodness" (sort off).

benjay10 committed:
Add OP consumer service

benjay10 committed:
Migrations: move to folders per year

benjay10 committed:
Migrations: flush previously exported by Loket

benjay10 committed:
Consumer SPARQL mappings for OP Public data

benjay10 committed:
Fix typo in PREFIX

benjay10 committed:
Remove unneeded Address consumer mapping

benjay10 committed:
Fix docker volume mount

benjay10 committed:
Add missing Bestuursorgaan In Tijd consumer mapping

claire-lovisa reviewed (changes requested):
We discovered recently issues and inconsistencies related to how the delta flows were implemented. I'm setting here a small list of the things that will have an impact on your PR:
- Move the consumer mapping to a `delta-consumers` folder to keep things similar to the other applications (we reworked that recently)
- `prov:Location` are now produced by OP as well so that can be flushed / re consumed in LDB too
- Adapt the mappings to avoind consumers deleting too much data when receiving a `delete` instruction, see https://github.com/lblod/app-mandatendatabank/pull/14 (mostly, all the mapping files where you `CONSTRUCT` both some `?p ?o` and some extra triples, like a extra type)

Sorry for the extra work, things are moving a bit lately

benjay10 committed:
Consumer config to own directory

benjay10 committed:
Location added to consumer mappings

benjay10 commented:
Working on your remarks: it seems that `prov:Location` is so new in the producer of OP that there is no data to consume yet. This is okay. There is also no data in Leidinggevenden app about `prov:Location` so there is no need for a migration to remove data.

benjay10 committed:
Rewrite for more correct deletes

Thanks to a warning from @claire-lovisa, we needed to write the mappings
to support deletes that don't remove new mapped type.

benjay10 commented:
I think I was able to correct the mappings now, keeping in mind that deletes are a bit weird.

claire-lovisa commented on config/delta-consumers/op-consumer/sparql-mappings/Location.sparql:
```diff
@@ -0,0 +1,14 @@
+    ?p ?o .
+
+  FILTER (?p in (
+    <http://mu.semte.ch/vocabularies/core/uuid>,
```
The type is missing here

claire-lovisa reviewed (changes requested):
Except for the minor comment, I tested the sync (initial + delta) and it works just fine :)

For the need to add the sparql parser, there is a trick for the initial sync to make it go a  LOT faster with the current version of the database:
```
  op-public-consumer:
    environment:
      DCR_LANDING_ZONE_DATABASE: "virtuoso"
      DCR_REMAPPING_DATABASE: "virtuoso"
```
The only drawback is that you might need to restart the cache after the initial sync but it's not too bad. For the delta sync we put it back to `database`

benjay10 committed:
Fix missing type in mapping

claire-lovisa committed:
Fix typo

benjay10 commented:
Oh my, sorry @claire-lovisa for the typo. I was being too fast again. :disappointed:
